docs/sphinx/sections/api/apidocs/io-managers.rst

```diff
@@ -46,6 +46,16 @@ Built-in IO Managers
   :annotation: IOManagerDefinition
 
 
+Input Managers (Experimental)
```

Comment:
I think these are stable, not experimental. @jamiedemaria and/or @OwenKephart can confirm.

Reply:
I haven't had much time to look into this, but i recently ran across  a funky edge case where I'm concerned that input managers don't cover the same functionality as root input managers. here's a (somewhat) trivial run down:

if we have op_a return 2.  op_b takes it as input, adds 1 and returns 
if we use a root input manager to provide 3 as input to op_b and just execute op_b we get 4 as expected. 

the typical case for using an input manager is to load the input to an op in a special way (ie output from an op is a dataframe, but we want it in a different format for the next op). 
if we have an input manager on op_b that multiplies the output by 2, we execute the job and get 5 as expected. 

the other use case for input managers is to replace root input managers
if we have a new input manager that provides 3 as input to op_b and we execute op_b we get 4 as expected. BUT if we execute the full job, the input manager will still provide 3 to op_b. this is because there isn't a good way to determine which type of input manager (the first version or the root manager replacing version) is intended

the way we get around this is by writing logic in the input manager to check if there is an upstream output, and if so we handle it differently (the example in the docs is to make the input manager extend from an IO manager and then call the IO manager's load_input function) 

this works pretty well except for the in-memory io manager because we don't have a way to access the instantiation of the in-mem io manager that is storing the output


I haven't had time to dig into this yet, but on the chance that addressing it causes the user facing API for input managers to shift a little bit, I think it should stay experimental.

That was a very long comment and a bit of a half baked example of the issue. I can write up an working code snippet tomorrow or hop on a call to explain if that would be helpful

Reply:
If we're unsure, best to keep it experimental now IMO, especially since we aren't getting rid of type loaders or materializers yet. We can always un-experimentalize later with little to no cost.